test/fw/ptl/utils/plugins/ptl_test_loader.py

```diff
@@ -190,9 +193,35 @@ def prepareTestLoader(self, loader):
         old_loadTestsFromNames = loader.loadTestsFromNames
 
         def check_loadTestsFromNames(names, module=None):
-            rv = old_loadTestsFromNames(names, module)
+            if not self.testfiles:
+                self._du = DshUtils()
+                ptl_test_dir = self._du.which(exe='pbs_benchpress')
+                ptl_test_dir = ptl_test_dir.replace('bin/pbs_benchpress',
+                                                    'tests')
+                user_test_dir = os.environ.get("PTL_TESTS_DIR", None)
+                if user_test_dir is not None:
```

Comment:
as discussed offline please remove this if-else condition and create a new list with names and append ptl_test_dir and user_test_dir based on availability and in order.